samplicate.c

```diff
@@ -318,7 +316,7 @@ static int
 samplicate (ctx)
      struct samplicator_context *ctx;
 {
-  unsigned char fpdu[PDU_SIZE];
+  unsigned char fpdu[ctx->pdulen];
```

Comment:
At first I was worried about the portability of this.  When I developed in C (a long time ago), only GCC allowed such computed array sizes for automatic variables.  According to https://gcc.gnu.org/onlinedocs/gcc/Variable-Length.html it is supported by ISO C99.  I don't know how widely that is supported.  Hopefully all compilers that people are likely to use will understand this.